code/app/src/main/java/com/example/goldencarrot/data/model/waitlist/LotterySystem.java

```diff
@@ -0,0 +1,103 @@
+package com.example.goldencarrot.data.model.waitlist;
+import com.example.goldencarrot.data.db.WaitListRepository;
+import com.example.goldencarrot.data.model.user.UserImpl;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
```

Comment:
This is a perfect example of a WaitList Controller which we need desperately. Please add this to controller package

Reply:
Please rename this to WaitListController.java